src/Go/boardAnalysis/goAI.ts

```diff
@@ -785,8 +797,9 @@ function getMoveOptions(
         ? defendMove
         : null;
     },
-    eyeMove: async () => (endGameAvailable ? null : getEyeCreationMove(board, player, availableSpaces) ?? null),
```

Comment:
These really shouldn't be async in the first place. The correct fix is to fix that. If you don't want to fix the typing, the next-most-preferable fix is to just disable the lint here.

I see two possible ways to fix the typing that are hopefully easy, either extend the explicit type up above, or (better) remove the explicit type entirely, and put `const` at the *end* of this object literal definition. That will cause TS to infer the record type specifically and directly from the members. 

An aside: because of the syntax, it's easy to think that making a function `async` is a simple change. However, the reality is that it's a very *big* change, with significant (but subtle) implications for state consistency and invariants. Ideally, very few of these functions would be async.

Reply:
All of the move option getters are awaited when they are referenced abstractly on line 810; making them all return a promise made typing much cleaner, since they are intended to be interchangeable.

I've refactored all of these to limit the async to just the few that are particularly slow (and interact with bonus time). This results in a significant amount of code change, but in my testing everything seems to work as expected after this refactor.